### PR TITLE
Add useBucketList hook to manage bucket list items

### PR DESCRIPTION
**Pull request by sarowarhosen03**
#5 
now bucket list items sync with  local storage
here im made up an custom hooks help of  using  createEffect  from solid.js to  sync the bucket items with localStorage more efficiently

### CODE REVIEW

File: src/App.jsx [hunk 1]
```diff
@@ -1,12 +1,10 @@
-import { createSignal } from 'solid-js';
+import { createEffect, createSignal } from 'solid-js';
 import { BucketListItem } from './BucketListItem';
 import { AddToBucket } from './AddToBucket';
+import useBucketList from './hooks/useBucketList';
 
 function App() {
-  const [items, setItems] = createSignal([
```
Comment (sarowarhosen03) [inline src/App.jsx:6]: shift to useBucketlist  hook .

**Review by sarowarhosen03 — COMMENTED** [review]
I think the implementation is good. Waiting for maintainer review

**Comment by atapas** [comment]
@sarowarhosen03 I liked what you did in this PR.

Could you please take the latest code, resolve merge conflicts and then change things according to what went in for the localstorage implementation with signal?

**Comment by sarowarhosen03** [comment]
> @sarowarhosen03 I liked what you did in this PR.
> 
> Could you please take the latest code, resolve merge conflicts and then change things according to what went in for the localstorage implementation with signal?

Yes I can do it give me some time i will update my pr

**Comment by atapas** [comment]
> > @sarowarhosen03 I liked what you did in this PR.
> > Could you please take the latest code, resolve merge conflicts and then change things according to what went in for the localstorage implementation with signal?
> 
> Yes I can do it give me some time i will update my pr

Cool... take ur time.. but look like u closed it. Also please create an issue for it.

**Comment by sarowarhosen03** [comment]
> @sarowarhosen03 I liked what you did in this PR.
> 
> Could you please take the latest code, resolve merge conflicts and then change things according to what went in for the localstorage implementation with signal?

Sir  should i take dummy bucket list item as initial state. Or i will remove it and start with empty . I think Starting with empty is good one. What's your opinion

**Comment by sarowarhosen03** [comment]
> > > @sarowarhosen03 I liked what you did in this PR.
> > > Could you please take the latest code, resolve merge conflicts and then change things according to what went in for the localstorage implementation with signal?
> > 
> > Yes I can do it give me some time i will update my pr
> 
> Cool... take ur time.. but look like u closed it. Also please create an issue for it.

I discard my changes and sync with your master branch I will create a new PR with my updated code soon

**Pull request closed without merging**